### ISSUE TITLE
AxesBackend should not handle permission checks

### ISSUE THREAD

**Issue by jcgiuffrida** [jazzband/django-axes#812]
### Expected behavior
AxesBackend raises `PermissionDenied` if a user is locked out. Otherwise, the other `AUTHENTICATION_BACKENDS` do the rest of the authentication work.

### Actual behavior
AxesBackend handles permissions checks just like Django's `contrib.auth.backends.ModelBackend` because it inherits from it. 

### Explanation
My site uses a custom authentication backend that handles directly-assigned user permissions differently (depending on certain conditions) than Django's contrib.auth.backends.ModelBackend. It took me a while to figure out that AxesBackend was actually messing with this, because it subclasses from ModelBackend, so that it (silently) runs all the methods of ModelBackend except for the `authenticate` method that it overrides. 

For example, suppose that we don't want to use Django's default logic of having superusers inherit all permissions. This is done in ModelBackend._get_permissions. Even if we override that or change it, AxesBackend still gives superusers all permissions when it runs that method. 

I do not believe this is expected or desired behavior, for the small subset of users like me who have a different setup and now have to override AxesBackend to change its inheritance or null out the permission-related methods. In my understanding, Axes is about authentication, not permissions, and does not even need to subclass from ModelBackend at all (as long as it has the [required backend methods](https://docs.djangoproject.com/en/3.2/topics/auth/customizing/#writing-an-authentication-backend)). 

### Proposed fix
I think it should inherit from `object` instead, and we should add a blank `get_user` method that returns `None` to conform with Django's requirements for custom backends. No change to the `authenticate` method, so no change to the Axes part of AxesBackend. However, AxesBackend would no longer run permissions checks because it would no longer inherit from ModelBackend. 

I'm happy to submit a PR to do this, but wanted to check as this is pretty sensitive stuff.

* Is my interpretation correct?
* The [documentation](https://django-axes.readthedocs.io/en/latest/2_installation.html) specifies that ModelBackend should come after AxesBackend, so the majority of users are probably doing that and would not be affected by this. What about users who do **not** currently use ModelBackend - perhaps because they have custom authentication logic in a separate backend - and are relying on AxesBackend to perform permissions checks? They may not even realize it. Perhaps instead of changing AxesBackend we could add a new backend class that inherits from `object` and leave the current one alone - and update the documentation with a note for users with custom authentication backends.

**Comment by aleksihakli** [comment]
Hi @lucaluca, and thanks for submitting the detailed issue description.

I agree that the design for the authentication backend is less than ideal, but how would the user lockouts be handled if not in the authetnication backend?

The documentation has the architectural image describing the login process and how the Django and Django Axes internals handle the user permission checks and lockouts:

https://django-axes.readthedocs.io/en/latest/7_architecture.html

**Comment by dwasyl** [comment]
@aleksihakli I just ran into the same problem that @jcgiuffrida ran into as well. I've developed a custom permission handler for some behind the scenes mutli-tenant stuff.

I believe with his original proposal, `django-axes` would still provide an authentication backend, however it would be limited in scope and *not* contain all of the user/group permission checks that are unnecessary. It means that `django-axes` couldn't be the only `AUTHENTICATION_BACKEND` specified, but that is not the recommendation from the docs anyway.

Another example of a limited scope backend, is the authentication backend provided by `django-guardian`, the `ObjectPermissionBackend` provided by that library [https://github.com/django-guardian/django-guardian/blob/devel/guardian/backends.py](https://github.com/django-guardian/django-guardian/blob/devel/guardian/backends.py), only includes the methods required of object-based permissions.

If the `axes` backend inherited from `object` (as suggested), a couple of additional methods would need to be specified, but a lot of unnecessary inherited code would be eliminated. It wouldn't impact the performance of `axes` at all, but if someone didn't have any other `BACKEND` specified, it would certainly break. The potential for a breaking change is problematic, but that would be for non-standard installs. For others, like me, it would stop `axes` from overstepping.

Anyway, my 2 cents if @jcgiuffrida already has a customized backend it would be great to see it contributed (especially so I don't need to also write my own)!

**Comment by aleksihakli** [comment]
If you have an implementation that you think is an improvement over the current one and you'd like to see it upstreamed for Axes the project is 100% open for contributions, please feel free to open a PR. The only criteria I'd like the implementation to fill is that the current backwards compatibility is kept if the proposal overrides the current implementation :)

**Comment by jcgiuffrida** [comment]
I opened a PR for this. In the meantime @dwasyl I just use a very basic fix in my project:

```
class MyAxesBackend(AxesBackend):
    """
    Override the AxesBackend to not check permissions.
    """
    def get_all_permissions(self, user_obj, obj=None):
        return set()
```

You may need to copy/paste other methods from `django.contrib.auth.backends.BaseBackend` for your project.

**Comment by aleksihakli** [comment]
Released as part of 5.36.0
